test_helpers_end_to_end_ng/src/mini_cluster.rs

```diff
@@ -41,34 +51,78 @@ impl MiniCluster {
+
+        Self {
+            router2,
+            ingester,
+            querier,
+            compactor,
+            other_servers: vec![],
+
+            org_id,
+            bucket_id,
+            namespace,
+        }
+    }
+
+    /// Create a "standard" shared MiniCluster that has a router, ingester,
     /// querier
     ///
-    /// Long term plan is that this will be shared across multiple tests if possible
-    pub async fn create_standard(database_url: String) -> Self {
-        let router2_config = TestConfig::new_router2(&database_url);
-        // fast parquet
-        let ingester_config =
-            TestConfig::new_ingester(&router2_config).with_fast_parquet_generation();
-        let querier_config = TestConfig::new_querier(&ingester_config);
+    /// Note: Since the underlying server processes are shared across multiple
+    /// tests so all users of this MiniCluster should only modify
+    /// their namespace
+    pub async fn create_shared(database_url: String) -> Self {
+        let start = Instant::now();
+        let mut shared_servers = GLOBAL_SHARED_SERVERS.lock().await;
+        debug!(mutex_wait=?start.elapsed(), "creating standard cluster");
 
-        // Set up the cluster  ====================================
-        Self::new()
-            .with_router2(router2_config)
-            .await
-            .with_ingester(ingester_config)
-            .await
-            .with_querier(querier_config)
-            .await
+        // try to reuse existing server processes
+        if let Some(shared) = shared_servers.take() {
+            if let Some(cluster) = shared.creatable_cluster().await {
+                debug!("Reusing existing cluster");
+
+                // Put the server back
+                *shared_servers = Some(shared);
+                let start = Instant::now();
+                // drop the lock prior to calling create() to allow
+                // others to proceed
+                std::mem::drop(shared_servers);
+                let new_self = cluster.create().await;
+                info!(total_wait=?start.elapsed(), "created new new mini cluster from existing cluster");
+                return new_self;
+            } else {
+                info!("some server proceses of previous cluster have already returned");
+            }
+        }
+
+        // Have to make a new one
+        info!("Create a new server");
+        let new_cluster = Self::create_non_shared_standard(database_url).await;
+
+        // Update the shared servers to point at the newly created server proesses
+        *shared_servers = Some(SharedServers::new(&new_cluster));
+        new_cluster
     }
 
-    /// return a "standard" MiniCluster that has a router, ingester,
-    /// querier and quickly persists files to parquet
-    pub async fn create_quickly_persisting(database_url: String) -> Self {
+    /// Create a non shared "standard" shared MiniCluster that has a router, ingester,
```

Comment:
I think there's an extra "shared" in here?

```suggestion
    /// Create a non shared "standard" MiniCluster that has a router, ingester,
```